src/mokka/src/integration-test/java/pl/hycom/mokka/emulator/mock/MockConfigurationControllerIT.java

```diff
@@ -0,0 +1,179 @@
+
+        Response response = RestAssured.get("/__admin/mappings/" + uuid);
+        expect().statusCode(200).contentType(ContentType.JSON).response();
+
+        Assertions.assertEquals(response.jsonPath().get("name"), name);
+        Assertions.assertEquals(response.jsonPath().get("request.url"), slash + path);
+        Assertions.assertEquals(response.jsonPath().get("request.method"), httpMethod);
+
+        mvc.perform(asyncRequest(delete("/configuration/" + id)).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void update_mock() throws Exception {
+
+        addMock();
+        String testChange = "test_change";
+
+        String requestBody = "{\"status\":" + status + ",\"httpMethod\":\"" + httpMethod + "\",\"path\":\""+path+"\",\"id\":"+id+",\"description\":\"test\",\"pattern\":null,\"name\":\"" + testChange + "\",\"enabled\":true,\"proxyBaseUrl\":null,\"timeout\":0,\"order\":0,\"configurationContent\":{\"string\":{\"value\":\"d\"}},\"updated\":\"2020-03-18T01:44:39.089+0000\",\"errors\":null,\"changes\":null,\"showChanges\":false,\"showConfiguration\":true,\"type\":\"string\"}";
+
+        mvc.perform(asyncRequest(put("/configuration")).contentType(MediaType.APPLICATION_JSON).with(csrf())
+            .content(requestBody))
+            .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings/" + uuid);
+        expect().statusCode(200).contentType(ContentType.JSON).response();
+
+        Assertions.assertEquals(response.jsonPath().get("name"), testChange);
+        Assertions.assertEquals(response.jsonPath().get("request.url"), slash + path);
+        Assertions.assertEquals(response.jsonPath().get("request.method"), httpMethod);
+
+        mvc.perform(asyncRequest(delete("/configuration/" + id)).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void disable_mock() throws Exception {
+
+        addMock();
+
+        mvc.perform(asyncRequest(post("/configuration/" + id +"/disable")).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings/" + uuid);
+        expect().statusCode(404).contentType(ContentType.JSON).response();
+
+        mvc.perform(asyncRequest(delete("/configuration/" + id)).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+
+        Assertions.assertEquals(response.getStatusCode(), HttpStatus.NOT_FOUND.value());
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void enable_mock() throws Exception {
+
+        addMock();
+
+        mvc.perform(asyncRequest(post("/configuration/"+ id +"/disable")).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
```

Comment:
I would rather propose to create disabled mock in one shot instead of creating enabled and then disabling it